File: 백준/Silver/10845. 큐/큐.py
import sys
import queue
input=sys.stdin.readline
q=queue.Queue()
N=int(input().rstrip())
for _ in range(N):
  M=input().rstrip()
  if("push" in M):
    q.put(int(M[5:]))
  elif("pop" in M):
    if(q.qsize()!=0):
      print(q.get())
    else:
      print(-1)
  elif(M=="size"):
    print(q.qsize())
  elif(M=="empty"):
    if(q.qsize()!=0):
      print(0)
    else:
      print(1)
  elif(M=="front"):
    if(q.qsize()!=0):
      print(q.queue[0])
    else:
      print(-1)
  elif(M=="back"):
    if(q.qsize()!=0):
      print(q.queue[-1])
    else:
      print(-1)

File: 백준/Silver/10845. 큐/큐.java
import java.util.*;

public class Main {
    public static void main(String[]args){
        Queue<Integer> q= new LinkedList<>();
        Scanner scan= new Scanner(System.in);
        int N= scan.nextInt();
        int lastNum=0;
        for(int i=0;i<N; i++){
            String M= scan.next();
            if(M.equals("push")){
                int O= scan.nextInt();
                q.offer(O);
                lastNum=O;
            }
            else if(M.equals("pop")){
                if (q.isEmpty()) {
                    System.out.println(-1);
                } else
                    System.out.println(q.poll());
            }
            else if(M.equals("size")){
                System.out.println(q.size());
            }
            else if (M.equals("empty")) {
                if (q.isEmpty()) {
                    System.out.println(1);
                }
                else {
                    System.out.println(0);
                }
            }
            else if (M.equals("front")) {
                if (q.isEmpty()) {
                    System.out.println(-1);
                } else
                    System.out.println(q.peek());
            }
            else if (M.equals("back")) {
                if (q.isEmpty()) {
                    System.out.println(-1);
                } else
                    System.out.println(lastNum);
            }
        }
    }
}